.github/workflows/functional_test.yaml

```diff
@@ -4,11 +4,18 @@ on: pull_request
 
 jobs:
   e2e-test:
-    name: ${{ matrix.container_runtime }}
+    name: Kubernetes ${{ matrix.kubernetes_version }} ${{ matrix.container_runtime }}
     runs-on: ubuntu-20.04
     strategy:
       fail-fast: false
       matrix:
+        # TODO: Automate updating this
+        kubernetes_version:
+          - v1.28.0 # Support: 28 Aug 2024 - 28 Oct 2024
+          - v1.27.4  # Support: 28 Apr 2024 - 28 Jun 2024
+          - v1.26.7 # Support: 28 Dec 2023 - 28 Feb 2024
+          - v1.25.12 # Support: 27 Aug 2023 - 27 Oct 2023
+          - v1.24.16  # Support: 28 May 2023 - 28 Jul 2023
```

Comment:
If support is past the limit on this, should we still test it?

Reply:
It is a nice to have and a gray area. These support dates come from [core Kuberenetes](https://endoflife.date/kubernetes). Other distributions like AKS, EKS, and GKE might support core Kubernetes versions longer than core Kubernetes does. I'll see if I can add more comment context to this or remove the oldest version from the testing coverage.

Reply:
Yeah we should at least these many kubernetes_version entries for now. I am adding automation for updating the kubernetes_version matrix entries soon that will expand and contract the entry count as needed.

Added more notes for contextext.